src/components/GoodsList/GoodsList.tsx

```diff
@@ -0,0 +1,175 @@
+import { range } from 'lodash';
+import { Button } from 'react-bootstrap';
+import { Component } from 'react';
+import { SortType } from '../../enums/SortType';
+
+interface Props {
+  goods: string[];
+}
+
+interface State {
+  isReversed: boolean;
+  sortBy: SortType;
+  minLength: number,
+  maxLength: number,
+  selectedLength: number,
+}
+
+export class GoodsList extends Component<Props, State> {
+  state = {
+    isReversed: false,
+    sortBy: SortType.Default,
+    minLength: 1,
+    maxLength: 10,
+    selectedLength: 1,
+  };
+
+  handleReverseClick = () => {
+    this.setState({ isReversed: true });
+  };
+
+  handleSortByLengthClick = () => {
+    this.setState({ sortBy: SortType.ByLength });
+  };
+
+  handleSortAlphabeticallyClick = () => {
+    this.setState({ sortBy: SortType.Alphabetical });
+  };
+
+  handleResetClick = () => {
+    const { minLength } = this.state;
+
+    this.setState({
+      isReversed: false,
+      sortBy: SortType.Default,
+      selectedLength: minLength,
+    });
+  };
+
+  handleSelectorChange = (value: string) => {
+    this.setState({
+      selectedLength: +value,
```

Comment:
Краще явно приводити до числа
```suggestion
      selectedLength: Number(value),
```